envisage/extension_point_binding.py

```diff
@@ -23,15 +23,15 @@ class ExtensionPointBinding(HasTraits):
     #### 'ExtensionPointBinding' interface ####################################
 
     # The object that we are binding the extension point to.
-    obj = Any
+    obj = WeakRef
```

Comment:
With this change, `self.obj` might be `None` (and might become `None` at unpredictable moments). So we'd need to update code that uses `self.obj` to allow for that, for example in `_set_trait` and `update_trait`.